api/image_pipeline/pipeline_image_util.py

```diff
@@ -0,0 +1,660 @@
+
+
+def get_delivery_repo_id(name: str) -> str:
+    """
+    Function to get the delivery repo id. Used to construct the delivery repo URL to direct to its page in Pyxis.
+
+    :name: Delivery repo name
+    """
+    url = f"https://pyxis.engineering.redhat.com/v1/repositories?filter=repository=={name}"
+    response = request_with_kerberos(url)
+
+    if response.status_code == 404:
+        raise exceptions.DeliveryRepoUrlNotFound(f"Couldn't find delivery repo link on Pyxis")
+
+    try:
+        repo_id = response.json()['data'][0]['_id']
+    except Exception:
+        raise exceptions.DeliveryRepoIDNotFound(f"Couldn't find delivery repo ID on Pyxis for {name}")
+
+    return repo_id
+
+
+# Methods
+@util.refresh_krb_auth
+def request_with_kerberos(url: str) -> requests.Response():
+    # Kerberos authentication
+    kerberos_auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL)
+
+    # Sending the kerberos ticket along with the request
+    response = requests.get(url, auth=kerberos_auth)
+
+    if response.status_code == 401:
+        raise exceptions.KerberosAuthenticationError("Kerberos Authentication failed")
+    if response.status_code == 403:
+        raise exceptions.AccessDenied("Errata Access Denied")
+
+    return response
+
+
+def get_image_stream_tag(distgit_name: str, version: str) -> str:
+    """
+    Function to get the image stream tag if the image is a payload image.
+    The for_payload flag would be set to True in the yml file
+
+    :distgit_name: Name of the distgit repo
+    :version: OCP version
+    """
+    url = f"https://raw.githubusercontent.com/openshift/ocp-build-data/openshift-{version}/images/{distgit_name}.yml"
+    response = requests.get(url)
+
+    yml_file = yaml.safe_load(response.content)
+    if yml_file.get('for_payload', False):  # Check if the image is in the payload
+        tag = yml_file['name'].split("/")[1]
+        return tag[4:] if tag.startswith("ose-") else tag  # remove 'ose-' if present
+
+
+# @util.cached
+def doozer_github_distgit(version: str) -> list:
+    """
+    Function to get the distgit to GitHub mappings from doozer
+
+    :version: OCP version
+    """
+    output = util.cmd_gather(
+        f"doozer --disable-gssapi -g openshift-{version} images:print --short '{{name}}: {{upstream_public}}'")
+    if "koji.GSSAPIAuthError" in output[2]:
+        raise exceptions.KerberosAuthenticationError("Kerberos authentication failed for doozer")
+
+    result = []
+    for line in output[1].splitlines():
+        array = line.split(": ")
+        result.append(array)
+
+    return result
+
+
+def github_distgit_mappings(version: str) -> dict:
+    """
+    Function to get the GitHub to Distgit mappings present in a particular OCP version.
+
+    :version: OCP version
+    """
+    output = doozer_github_distgit(version)
+    dict_data = defaultdict(list)
+    for line in output:
+        if len(line) == 2:
+            dict_data[line[1].split("/")[-1]].append(line[0])
```

Comment:
an example would be helpful here, it's not obvious what the expected inputs and outputs look like

Reply:
Done!